src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```
@@ -873,6 +980,141 @@ def _are_dtypes_compatible_with_snowflake(cls, compiler: BaseQueryCompiler) -> b
                 return False
         return True
 
+    @classmethod
+    def _has_unsupported_args(
+        cls,
+        api_cls_name: Optional[str],
+        operation: Optional[str],
+        args: Optional[MappingProxyType[Any, Any]],
+    ) -> bool:
+        """
+        Check if method call contains unsupported args that require pandas backend.
+
+        This method evaluates registered UnsupportedArgsRule conditions to determine
+        if the provided arguments are not supported by Snowpark pandas and should
+        trigger an auto-switch to pandas backend.
+
+        Args:
+            api_cls_name: Class name (DataFrame, Series, BasePandasDataset, None for top-level functions)
+            operation: Method name
+            args: Method arguments including args
+
+        Returns:
+            True if unsupported args detected and auto-switch should occur
+        """
+        if not operation:
+            return False
+
+        if args is None:
+            return False
+
+        rule = HYBRID_SWITCH_FOR_UNSUPPORTED_ARGS.get(
+            MethodKey(api_cls_name, operation)
+        )
+        if rule is None:
+            return False
+
+        # Check custom conditions
+        for condition in rule.unsupported_conditions:
+            try:
+                # Validate condition format
+                if not isinstance(condition, tuple) or len(condition) != 2:
+                    logging.warning(
+                        f"Skipping malformed condition for {api_cls_name}.{operation}: "
+                        f"expected tuple of length 2, got {type(condition).__name__} "
+                        f"of length {len(condition) if hasattr(condition, '__len__') else 'unknown'}. "
+                        f"Condition: {condition}"
+                    )
+                    continue
```

IMO it would be better to 
1) fail loudly with an exception if this condition occurs
2) do the check after all decorators have been run, rather than at runtime during method invocation, OR use a helper method to insert entries into the rules dict that raises an exception for any malformed entries when they're added

The rule dictionary is initialized after all the query compiler methods are declared and decorators are run, and never mutated again. As such, we can afford to do this validation either once after the decorators are all run, or while we're in the process of populating the dict in the first place. Raising a warning rather than failing loudly is harder to detect and triage.